File: WebApplication1/Models/Band.cs
using Models;
using System;

namespace WebApplication1.Models
{
    public class Band
    {
        public int BandId { get; set; }
        public string Name { get; set; }
        public string SchoolName { get; set; }
        public string Location { get; set; }
        public int NumberOfMembers { get; set; }
        public DateTime CreatedAt { get; set; }

        // Navigation Properties
        public ICollection<Recruiter>? Recruiters { get; set; } = new List<Recruiter>();
        public ICollection<Offer>? Offers { get; set; } = new List<Offer>();
        public ICollection<InterestedStudent> InterestedStudents { get; set; } = new List<InterestedStudent>();
    }
}

File: client/src/models/Band.ts
import Recruiter  from './Recruiter';
import  Offer  from './Offer';
import Interest  from './Interest';

export default interface Band {
    bandId: number;
    name: string;
    schoolName: string;
    location: string;
    numberOfMembers: number;
    createdAt: string;
    recruiters?: Recruiter[]; // List of recruiters associated with this band
    offers?: Offer[]; // List of offers associated with this band
    interestedStudents?: Interest[]; // List of students interested in this band
}
